composeApp/src/commonMain/kotlin/org/ooni/probe/ui/settings/proxy/ProxyScreen.kt

```diff
@@ -0,0 +1,228 @@
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.selection.selectable
+import androidx.compose.foundation.selection.selectableGroup
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.automirrored.filled.ArrowBack
+import androidx.compose.material.icons.filled.ArrowDropDown
+import androidx.compose.material3.DropdownMenu
+import androidx.compose.material3.DropdownMenuItem
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.RadioButton
+import androidx.compose.material3.Text
+import androidx.compose.material3.TopAppBar
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.semantics.Role
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.unit.dp
+import ooniprobe.composeapp.generated.resources.Res
+import ooniprobe.composeapp.generated.resources.Settings_Proxy_Custom
+import ooniprobe.composeapp.generated.resources.Settings_Proxy_Custom_Hostname
+import ooniprobe.composeapp.generated.resources.Settings_Proxy_Custom_Port
+import ooniprobe.composeapp.generated.resources.Settings_Proxy_Custom_Protocol
+import ooniprobe.composeapp.generated.resources.Settings_Proxy_Enabled
+import ooniprobe.composeapp.generated.resources.Settings_Proxy_None
+import ooniprobe.composeapp.generated.resources.Settings_Proxy_Psiphon
+import ooniprobe.composeapp.generated.resources.back
+import org.jetbrains.compose.resources.StringResource
+import org.jetbrains.compose.resources.stringResource
+import org.ooni.probe.data.repositories.DOMAIN_NAME
+import org.ooni.probe.data.repositories.IPV6_ADDRESS
+import org.ooni.probe.data.repositories.IP_ADDRESS
+
+@Composable
+fun ProxyScreen(
+    onEvent: (ProxyViewModel.Event) -> Unit,
+    state: ProxyViewModel.State,
+) {
+    Column {
+        TopAppBar(
+            title = {
+                Text(stringResource(Res.string.Settings_Proxy_Enabled))
+            },
+            navigationIcon = {
+                IconButton(onClick = {
+                    if (state.proxyType != ProxyType.CUSTOM ||
+                        (isValidDomainNameOrIp(state.proxyHost ?: "") && isValidPort(state.proxyPort ?: 0))
+                    ) {
+                        onEvent(ProxyViewModel.Event.BackClicked)
```

Comment:
It feels weird to prevent the back button from working, without any error message.